src/modules/module_21500.c

```diff
@@ -0,0 +1,307 @@
+
+  return esalt_size;
+}
+
+u64 module_tmp_size (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  const u64 tmp_size = (const u64) sizeof (pbkdf2_sha1_tmp_t);
+
+  return tmp_size;
+}
+
+u32 module_pw_max (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  // this overrides the reductions of PW_MAX in case optimized kernel is selected
+  // IOW, even in optimized kernel mode it support length 256
+
+  const u32 pw_max = PW_MAX;
+
+  return pw_max;
+}
+
+char *module_jit_build_options (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra, MAYBE_UNUSED const hashes_t *hashes, MAYBE_UNUSED const hc_device_param_t *device_param)
+{
+  char *jit_build_options = NULL;
+
+  if (device_param->opencl_device_vendor_id == VENDOR_ID_AMD)
+  {
+    hc_asprintf (&jit_build_options, "-D NO_UNROLL");
+  }
+
+  return jit_build_options;
+}
+
+int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED void *digest_buf, MAYBE_UNUSED salt_t *salt, MAYBE_UNUSED void *esalt_buf, MAYBE_UNUSED void *hook_salt_buf, MAYBE_UNUSED hashinfo_t *hash_info, const char *line_buf, MAYBE_UNUSED const int line_len)
+{
+  u64 *digest = (u64 *) digest_buf;
+
+  pbkdf2_sha1_t *pbkdf2_sha1 = (pbkdf2_sha1_t *) esalt_buf;
+
+  token_t token;
+
+  token.token_cnt  = 3;
+
+  token.signatures_cnt    = 1;
+  token.signatures_buf[0] = SIGNATURE_SOLARWINDS_ORION;
+
+  token.len[0]     = 14;
+  token.attr[0]    = TOKEN_ATTR_FIXED_LENGTH
+                   | TOKEN_ATTR_VERIFY_SIGNATURE;
+
+  token.sep[1]     = '$';
+  token.len_min[1] = 0;
+  token.len_max[1] = 16;
+  token.attr[1]    = TOKEN_ATTR_VERIFY_LENGTH;
+
+  token.sep[2]     = '$';
+  token.len_min[2] = 64;
+  token.len_max[2] = 256;
+  token.attr[2]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_BASE64A;
+
+  const int rc_tokenizer = input_tokenizer ((const u8 *) line_buf, line_len, &token);
+
+  if (rc_tokenizer != PARSER_OK) return (rc_tokenizer);
+
+  // iter
+
+  const u32 iter = 1000;
+
+  salt->salt_iter = iter - 1;
+
+  // salt
+
+  char *salt_pos = (char *) token.buf[1];
+  int salt_len = token.len[1];
+
+  if (salt_len > 16) return (PARSER_SALT_LENGTH);
+
+  char custom_salt[17];
+
+  memset (custom_salt, 0, sizeof (custom_salt));
+
+  for (int i = 0; i < salt_len; i++, salt_pos++)
+  {
+    if (*salt_pos >= 'A' && *salt_pos <= 'Z') {
+      custom_salt[i] = *salt_pos + 32;
+    } else {
+      custom_salt[i] = *salt_pos;
+    }
+  }
+
+  if (salt_len < 8)
+  {
+    strncat (custom_salt, SALT_PADDING, 8 - salt_len);
+    salt_len = 8;
+  }
+
+  salt_pos = custom_salt;
+
+  memcpy (pbkdf2_sha1->salt_buf, salt_pos, salt_len);
+
+  salt->salt_len = salt_len;
```

Comment:
FYI, if you are setting 16 (or even 20) bytes in the salt->salt_buf buffer, then you need to set salt->salt_len to 16 (or 20), too. The reason is that hashcat will only compare data of length salt->salt_len in order to detect duplicate salts. If you are OK with this you can leave it as it is.